src/librustc_typeck/diagnostics.rs

```diff
@@ -3426,6 +3426,47 @@ parameters. You can read more about it in the API documentation:
 https://doc.rust-lang.org/std/marker/struct.PhantomData.html
 "##,
 
+E0436: r##"
+Functional update syntax can only be used on struct types.
+Erroneous code example:
+
+```compile_fail
+enum Foo { A { x: u32, y: u32 } }
+
+fn bar() {
+    baz(Foo::A { x: 5, y: 10 });
+}
+
+fn baz(Foo f) -> Foo {
+    Foo::A { x: 6, ..f } // error: functional record update syntax requires
+                         //        a struct
+}
+```
+
+The functional update syntax may only be used on struct types. It is an error
+to use it on other types such as enums. To fix this, you will need to manually
+initialize the value. For enums, you should match on the variants and copy the
+values to the new enum value:
+
+```
+enum Foo { A { x: u32, y: u32 } }
+
+fn bar() {
+    baz(Foo:A { x: 5, y: 10 });
+}
+
+fn baz(Foo f) -> Foo {
+    match f {
+        //match each variant of the enum and bind each value to a name
```

Comment:
Nits: space after `//` in comments; start with capital letter, end with period (i.e., use sentence structure)